org.freecad.FreeCAD.yaml

```diff
@@ -0,0 +1,801 @@
+    config-opts:
+      - -DBUILD_SHARED_LIBS=ON
+    sources:
+      - type: git
+        url: https://github.com/fmtlib/fmt.git
+        commit: a33701196adfad74917046096bf5a2aa0ab0bb50 # tag 9.1.0
+
+  - name: yaml-cpp
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DBUILD_SHARED_LIBS=ON
+    sources:
+      - type: git
+        url: https://github.com/jbeder/yaml-cpp
+        commit: f7320141120f720aecc4c32be25586e7da9eb978 # tag 0.8.0
+
+  - name: FreeCAD
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      - -DENABLE_DEVELOPER_TEST=OFF
+      - -DCMAKE_INSTALL_PREFIX=/app/freecad
+      - -DCMAKE_INSTALL_DATAROOTDIR=/app/share
+      # While redundant (this triggers a warning with the linter) removing this
+      # cause CMake to stop finding dependencies.
+      - -DCMAKE_PREFIX_PATH=/app
+      - -DSHIBOKEN_INCLUDE_DIR=/app/include/shiboken6
+      - -DPYSIDE_INCLUDE_DIR=/app/include/PySide6
+      - -DINSTALL_TO_SITEPACKAGES=ON
+      - -DFREECAD_USE_PYBIND11=ON
+    post-install:
+      - install -Dm755 ../openscad.sh ${FLATPAK_DEST}/bin/openscad
+      - ln -s /app/freecad/bin/FreeCADCmd /app/bin/FreeCADCmd
+      - ln -s /app/freecad/bin/FreeCADCmd /app/bin/freecadcmd
+      - ln -s /app/freecad/bin/FreeCAD /app/bin/FreeCAD
+      - ln -s /app/freecad/bin/FreeCAD /app/bin/freecad
+    sources:
+      - type: git
+        url: https://github.com/FreeCAD/FreeCAD.git
+        commit: 60a251354e203bdb924e2190fb9f5e18b48d7362
+      - type: patch
+        path: patches/OpenSCAD_transfermechanism.patch
+      - type: file
+        path: openscad.sh
+      - type: shell
+        commands:
+          # revcount needs to be updated any time the source commit or tag is changed
+          # to obtain the revcount for a given gitRef (commit hash, branch or tag) run
+          # echo $(($(curl -s 'https://api.github.com/repos/FreeCAD/FreeCAD/compare/120ca87015...'"$gitRef"'' | grep '"ahead_by":' | sed -s 's/ //g;s/"ahead_by"://;s/,//')+1))
+          - |
+            revcount=38642
+            sed -i 's|${PACKAGE_WCREF}|'"$revcount"' (Git)|' src/Build/Version.h.cmake
+            sed -i 's|@PACKAGE_VERSION@|@PACKAGE_VERSION@.'"$revcount"'|' src/XDGData/org.freecad.FreeCAD.metainfo.xml.in
+            sed -i 's|${PACKAGE_WCURL}|https://github.com/FreeCAD/FreeCAD.git|' src/Build/Version.h.cmake
+          - | # fix paths for flatpak environment
+            sed -i 's|/usr|/app|g' cMake/FindOCC.cmake src/Gui/GraphvizView.cpp
+            sitepackagesDir=$(python3 -c 'import sysconfig; print(sysconfig.get_paths()["platlib"])' | sed 's|/usr|/app|')
+            sed -i 's|${python_libs}|'"$sitepackagesDir"'|' src/Ext/freecad/CMakeLists.txt
+          # We need to change the doc icon name so it can be exported too
+          - sed -i "s|application-x-extension-fcstd|${FLATPAK_ID}.application-x-extension-fcstd|" src/XDGData/org.freecad.FreeCAD.xml src/Gui/CMakeLists.txt
```

Comment:
no. please use `rename-mime-icons`.

Reply:
docs said it was preferable to rename the source file but I don't mind, does `rename-mime-icons` go at the top level or in the freecad step? If I use `rename-mime-icons` do I still have to patch the xml file or that takes care of that too?

Reply:
like `rename-desktop-file` and `rename-icon`

it will rename the icons and fix the mime file.